checks.d/zk.py

```diff
@@ -96,6 +156,50 @@ def check(self, instance):
                 self.service_check('zookeeper.mode', status, message=message,
                                    tags=sc_tags)
 
+        if zk_version and LooseVersion(zk_version) > LooseVersion("3.4.0"):
+            try:
+                mntr_out = self._send_command('mntr', *cx_args)
+            except ZKConnectionFailure:
+                self.increment('zookeeper.timeouts')
+                self.report_instance_mode(hostname, 'down', tags)
+                raise
+            except:
+                e = sys.exc_info()[1]
+                print >> sys.stderr, "Error: %s" % e
+                self.increment('zookeeper.datadog_client_exception')
+                self.report_instance_mode(hostname, 'unknown', tags)
+                raise
+            else:
+                metrics, mode = self.parse_mntr(mntr_out)
+                mode_tag = "mode:%s" % mode
+                if mode != 'inactive':
+                    for name in metrics:
+                        self.gauge(name, metrics[name], tags=tags + [mode_tag])
+
+
+    def report_instance_mode(self, hostname, mode, tags):
+        tags = tags + ['mode:%s' % mode]
+        self.set('zookeeper.instances', hostname, tags=tags)
+
+        gauges = {
+            'leader': 0,
+            'follower': 0,
+            'observer': 0,
+            'standalone': 0,
+            'down': 0,
+            'inactive': 0,
+            'unknown': 0
+        }
+
+        if mode in gauges:
+            gauges[mode] = 1
+        else:
+            gauges['unknown'] = 1
```

Comment:
Maybe cleaner to rely on a class level variable `ZookeeperCheck.STATUS_TYPES` and then building the `gauges` dictionary from it. What do you think?